File: Python/Stack_array/stack.py
class Stack:
  def __init__(self, _array = []):
    self.array = _array
    self.top = len(_array) - 1

def pop(stack):
  temp = stack.top
  stack.top -= 1
  return stack.array[temp]

def push(stack, val):
  stack.top += 1
  stack.array[stack.top:] = [val]

File: C/Stack_array/stack.c
#include "stack.h"

Stack Stack_init() {
	Stack s;
	s.top = -1;
	return s;
}
void push(Stack *s, short val) {
	signed char *top = &(s->top);
	short *array = s->array;
	*top += 1;
	array[*top] = val;
}
short pop(Stack *s) {
	signed char *top = &(s->top);
	short *array = s->array;
	short temp = array[*top];
	*top -= 1;
	return temp;
}
